app/main.py

```diff
@@ -1,3 +1,52 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.car import Car
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as config:
+        config = json.load(config)
+        for customer in config["customers"]:
+            res = {}
+            print(f"{customer['name']} has {customer['money']} dollars")
+
+            for shop in config["shops"]:
+
+                car = Car(
+                    config["FUEL_PRICE"],
+                    customer["car"]["fuel_consumption"],
+                    customer["location"], shop["location"]
+                ).calculate_trip_cost()
+
+                sh = Shop(
+                    customer["name"],
+                    customer["product_cart"],
+                    shop["products"]
+                ).calculate_total_price()
+
+                res[car + sh] = [
+                    customer["name"],
+                    customer["product_cart"],
+                    shop["products"], shop["name"]
+                ]
+
+                print(f"{customer['name']}'s trip to the "
+                      f"{shop['name']} costs {car + sh}")
+
+            if customer["money"] < min(res.keys()):
+                print(f"{customer['name']} doesn't have enough "
+                      f"money to make a purchase in any shop")
+                break
+
+            print(f"{customer['name']} rides to {res[min(res)][-1]}\n")
```

Comment:
When the customer arrives at the shop his location should equal to shop location. The same when he arrives home.